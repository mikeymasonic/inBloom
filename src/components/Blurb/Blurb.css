.blurbContainer {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100vw;

  .Blurb {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 60%;
    text-align: justify;
    padding: 1.5em;
    margin-bottom: 2em;
    background: linear-gradient(120deg, #d3b1fb, #d6b1fb, #e0b3fb, #edb5fb, #fbb7fa, #fbbaea, #fcbcdb, #fcbfce, #fcc1c3, #fcccc3, #fcd3c5, #fcd6c5);
    box-shadow: 0px 0px 4px 4px #070E0D;
  }
}

@media (max-width: 1200px) {
  .blurbContainer {

    .Blurb {
      width: 70%;
      padding: 0.5em 1em 1em 1em;
    }
  }
}

@media (max-width: 667px) {
  .blurbContainer {

    .Blurb {
      width: 85%;
      padding: 0.5em 1em 1em 1em;
      font-size: 0.9em;
    }
  }

}

