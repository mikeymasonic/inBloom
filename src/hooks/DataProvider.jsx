import React, { createContext, useState, useContext } from 'react';
import PropTypes from 'prop-types';

const DataContext = createContext();

export const DataProvider = ({ children }) => {
  // const [song, setSong] = useState('../mp3/example2.mp3');
  const [song, setSong] = useState(null);
  const [mapLocation, setMapLocation] = useState(null);
  const [playerVisible, setPlayerVisible] = useState(false);
  const [visualizerDisplay, setVisualizerDisplay] = useState(false);

  const handleSong = (track) => {
    setSong(track);
    setMapLocation(track.mapLocation);
  };

  const handleMapLocation = (location) => {
    setMapLocation(location);
  };

  const handlePlayerVisible = (bool) => {
    setPlayerVisible(bool);
  };

  const handleVisualizerDisplay = (bool) => {
    setVisualizerDisplay(bool);
  };

  return (
    <DataContext.Provider
      value={{
        song,
        mapLocation,
        playerVisible,
        visualizerDisplay,
        handleSong,
        handleMapLocation,
        handlePlayerVisible,
        handleVisualizerDisplay,
      }}
    >
      {children}
    </DataContext.Provider>
  );
};

DataProvider.propTypes = {
  children: PropTypes.node,
};

// get state
export const useSong = () => {
  const { song } = useContext(DataContext);
  return song;
};

export const useMapLocation = () => {
  const { mapLocation } = useContext(DataContext);
  return mapLocation;
};

export const usePlayerVisible = () => {
  const { playerVisible } = useContext(DataContext);
  return playerVisible;
};

export const useVisualizerDisplay = () => {
  const { visualizerDisplay } = useContext(DataContext);
  return visualizerDisplay;
};

// get handlers
export const useHandleSong = () => {
  const { handleSong } = useContext(DataContext);
  return handleSong;
};

export const useHandleMapLocation = () => {
  const { handleMapLocation } = useContext(DataContext);
  return handleMapLocation;
};

export const useHandlePlayerVisible = () => {
  const { handlePlayerVisible } = useContext(DataContext);
  return handlePlayerVisible;
};

export const useHandleVisualizerDisplay = () => {
  const { handleVisualizerDisplay } = useContext(DataContext);
  return handleVisualizerDisplay;
};

