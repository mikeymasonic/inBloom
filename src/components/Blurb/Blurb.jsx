import React from 'react';
import Popup from 'reactjs-popup';
import { useMapLocation } from '../../hooks/DataProvider';
import { songs, aboutSite } from '../../utils/data';
import Upload from '../Upload/Upload';
import 'reactjs-popup/dist/index.css';
import styles from './Blurb.css';

const Blurb = () => {
  const mapLocation = useMapLocation();

  const blurbNodes = songs.map((song) => {
    if (mapLocation === song.mapLocation) {
      return <p key={song.title}>{song.description}</p>;
    }
  });

  return (
    <section className={styles.Blurb}>
      {blurbNodes}
      {!mapLocation && <p>{aboutSite.description}</p>} 
      {mapLocation && 
      <Popup 
        modal
        closeOnDocumentClick={true}
        trigger={open => (
          <button open={open}>Upload</button>
        )}>
        {close => (
          <Upload close={close} />
        )}
      </Popup>
      } 
    </section>
  );
};

export default Blurb;
