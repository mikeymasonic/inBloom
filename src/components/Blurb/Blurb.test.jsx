import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { DataProvider } from '../../hooks/DataProvider';
import Blurb from './Blurb';

describe('Player component', () => {
  afterEach(() => cleanup());
  it('renders Player', () => {
    const { asFragment } = render(
      <Router>
        <DataProvider>
          <Blurb />
        </DataProvider>
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
