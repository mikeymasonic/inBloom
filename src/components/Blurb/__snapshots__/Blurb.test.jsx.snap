// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player component renders Player 1`] = `
<DocumentFragment>
  <section
    class="Blurb"
  >
    <p>
      This is a site description for home page.  Lorem ipsum dolor sit amet consectetur, adipisicing elit. Quia dolorem ullam, unde aliquam tenetur consequatur ut in esse harum sunt! Atque dolorem provident accusamus sint? Ducimus, magni. Id, enim non? Lorem ipsum, dolor sit amet consectetur adipisicing elit. Voluptatem animi ex repellendus eaque. Quam blanditiis cum quo sunt laudantium nihil earum unde dolore laborum! Id libero iusto similique minus dignissimos. Lorem ipsum dolor sit amet consectetur, adipisicing elit. Impedit dolorem ut omnis corrupti consequatur totam, aliquid ex qui cum deleniti minima, maiores temporibus doloribus, voluptatem et veniam eligendi vel? Distinctio.
    </p>
  </section>
</DocumentFragment>
`;
